src/main/java/map/BattleMap.java

```diff
@@ -17,11 +17,31 @@ public BattleMap(int size) {
     this.map.addAll(Collections.nCopies(size * size, 0));
   }
 
-  public void putShip(Point p) {
-    map.add(p.to1D(size), 1);
+  public List<Integer> getMap() {
+    return new ArrayList<>(map);
+  }
+
+  public boolean isBusy(Point point) {
+    return (getAt(point) == 1 || getAt(point) == 2);
+  }
+
+  public boolean isSunk(List<Point> list) {
+    return list.stream().allMatch(point -> getAt(point) == 2);
   }
 
   public Integer getAt(Point p) {
     return map.get(p.to1D(size));
   }
+
+  public void setCellState(Point point, States shipState) {
+    if (shipState.getValue() < getAt(point))
```

Comment:
I think you are trying too hard to express a simple idea. The top three `if`s should be removed. The last `if` is the one we need.  Furthermore, the cases we consider with the `if`s are mutually exclusive, however, when not using `else`, all of them will be checked.